lib/completion.ts

```diff
@@ -29,6 +30,10 @@ export interface CompletionInstance {
 
 export class Completion implements CompletionInstance {
   completionKey = 'get-yargs-completions';
+  // we need to handle the completion key in multiple formats because the
+  // `strip-dashed` parser option might remove the hyphenated version from the
+  // parsed arguments
+  completionKeys = [this.completionKey, 'getYargsCompletions'];
```

Comment:
The possibility of user-specified parser configurations like `skip-dashed` makes looking for options in argv rather painful! I did a local fix in #2308 for one case, but I am think we need a utility routine to make it convenient and robust to find options in argv. There are multiple places where it is potentially an issue, like here.